src/app/editor/primitive-field/primitive-field.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+
+import { SearchableDropdownComponent } from '../searchable-dropdown';
+
+import { SchemaValidationService } from '../shared/services';
+
+@Component({
+  selector: 'primitive-field',
+  directives: [SearchableDropdownComponent],
+  providers: [SchemaValidationService],
+  styles: [
+    require('./primitive-field.component.scss')
+  ],
+  template: require('./primitive-field.component.html')
+})
+export class PrimitiveFieldComponent {
+
+  @Input() value: string | number | boolean;
+  @Input() schema: Object;
+  error: Error;
+
+  @Output() onValueChange = new EventEmitter<any>();
+
+  constructor(private schemaValidationService: SchemaValidationService) {
+
+  }
+
+  onModelChange(event: any) {
```

Comment:
Here the parameter is not an `event` right? Shall we call it something like `value` ?